File: contracts/go/utils/Message/message.go
package message

import (
	"math/big"
	"os"
	"strings"

	common "qanx.space/qanx-contracts/go/utils/Common"
)

// Retrieves the sender of the transaction
func Sender() string {
	sender := strings.ToLower(os.Getenv("MSG_SENDER"))

	if len(sender) == 0 {
		os.Stderr.WriteString("Message: Sender is not known\n")
		os.Exit(1)
	}

	return sender
}

// Retrieves the QANX value of the transaction
func Value() *big.Int {
	n, _ := common.ParseBig256(os.Getenv("MSG_VALUE"))

	return n
}

File: contracts/js/utils/Message/message.js
module.exports.sender = function sender() {
  let sender = process.env.MSG_SENDER.toLowerCase();

  if (!sender) {
    process.stderr.write("Message: Sender is not known\n");
    process.exit(1);
  }

  return sender;
};

module.exports.value = function value() {
  let value = process.env.MSG_VALUE || "0";
  return BigInt(value);
};
